### PR TITLE
Added diff command to show changes to a service

### PR DESCRIPTION
**Pull request by thebsdbox**
DO NOT MERGE, requires a vendor update

```
./diver ucp services diff --name web
@ ["Mode","Replicated","Replicas"]
- 10
+ 5
@ ["TaskTemplate","ContainerSpec","Secrets"]
+ [{"File":{"GID":"0","Mode":292,"Name":"NewSecret","UID":"0"},"SecretID":"zljv3oz04b07ngan2sy40g5lh","SecretName":"NewSecret"},{"File":{"GID":"0","Mode":292,"Name":"NewSecret1","UID":"0"},"SecretID":"zljv3oz04b07ngan2sy40g5lh","SecretName":"NewSecret"}]
```
Shows i've modified the amount of replicas and added two secrets.

I've also added a `--pretty` option to render the JSON object changes human readable.

### CODE REVIEW

**Comment by thebsdbox** [comment]
If approved, can be merged as the vendoring has been completed.

File: pkg/ucp/ucpService.go [hunk 1]
```diff
@@ -141,3 +143,29 @@ func (c *Client) ReapFailedTasks(serviceName string, rmvol, kill bool) error {
 
 	return nil
 }
+
+//GetServiceDifference - This will compare a service spec and previous spec and output the differences
+func (c *Client) GetServiceDifference(serviceName string, pretty bool) (string, error) {
```
Comment (joeabbey) [inline pkg/ucp/ucpService.go:148]: Some debug statements in the checks below might be nice

**Review by joeabbey — APPROVED** [review]
Overall, LGTM.

I reset to head and cleaned and it builds on the first try! 👍

File: pkg/ucp/ucpService.go (hunk 1 above)
Comment (thebsdbox) [reply pkg/ucp/ucpService.go:148]: on it !

**Comment by thebsdbox** [comment]
@joeabbey sorted :-)

**Comment by joeabbey** [comment]
LGTM !! @johnharris85 any thoughts?

**Comment by thebsdbox** [comment]
@joeabbey he's on annual leave, I think this can probably be merged now :)

**Pull request merged**